dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/route/RouteService.java

```diff
@@ -126,12 +136,43 @@ public class RouteService {
           "last-modified",
           "etag");
 
+  private final List<String> allowedRouteRegexRemoteServers = new ArrayList<>();
+
   @PostConstruct
   public void postConstruct() {
+    String routeRemoteServersAllowed =
+        configuration
+            .getPropertyOrDefault(ConfigurationKey.ROUTE_REMOTE_SERVERS_ALLOWED, "")
+            .strip();
+    if (!routeRemoteServersAllowed.isEmpty()) {
+      for (String host : routeRemoteServersAllowed.split(",")) {
+        validateHost(host);
+        allowedRouteRegexRemoteServers.add(TextUtils.createRegexFromGlob(host));
+      }
+    }
+
     webClient = WebClient.builder().clientConnector(clientHttpConnector).build();
     dataBufferFactory = new DefaultDataBufferFactory();
   }
 
+  protected void validateHost(String host) {
+    if (!(host.startsWith("http://") || host.startsWith("https://"))) {
+      throw new IllegalStateException(
+          "Allowed route URL scheme must be either http or https: " + host);
+    }
+    String urlWithoutPortNo =
+        host.substring(0, host.lastIndexOf(":") > 5 ? host.lastIndexOf(":") : host.length());
+    URL url;
+    try {
+      url = new URL(urlWithoutPortNo);
```

Comment:
It looks like this works also with wildcard routes like `https://*.dhis2.org`, which is a little surprising to me but if that's the case then should be ok!